### ISSUE TITLE
Subscriptions show shipping selector in Checkout

### ISSUE THREAD

**Issue by crunnells** [ingenesis/shopp#3236]
This seems to happen when you've got a product with a subscription variant, and a regular shipped variant. Shipping isn't applied, BUT the shipping rate selector shows regardless. Confusing for customers for sure.

**Comment by jonathandavis** [comment]
The issue is that subscriptions are outside of product type really. This shows the limitation in the current system. What if they want to sell a subscription to a shipped magazine and want shipping charges collected?

The subscription support in Shopp today is implemented completely incorrectly. This is a symptom of it.

**Comment by crunnells** [comment]
Right, except in odd cases where you have a shipped variant alongside a subscription variant... If the customer chooses the shipped variant, the product gets moved into the `shipped` array, and never leaves it.

But I digress; you're absolutely right that this is a symptom of subscriptions not being supported correctly. But in the interest of fixing some of these things ahead of the subscription system being overhauled, I'm reporting issues like this so that I can look at them later and address them *now*.
